src/wh_client.c

```diff
@@ -513,9 +516,10 @@ int wh_Client_KeyCacheResponse(whClientContext* c, uint16_t* keyId)
     uint16_t action;
     uint16_t size;
     int ret;
-    whPacket packet[1] = {0};
+    whPacket* packet;
     if (c == NULL || keyId == NULL)
         return WH_ERROR_BADARGS;
+    packet = (whPacket*)c->comm->data;
```

Comment:
Use accessor function and check return value.
```suggestion
    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
```